packages/peregrine/lib/talons/Region/useRegion.js

```diff
@@ -1,21 +1,37 @@
+import { useEffect, useRef } from 'react';
 import { useQuery } from '@apollo/react-hooks';
-import { useFieldState } from 'informed';
+import { useFieldApi, useFieldState } from 'informed';
 
 export const useRegion = props => {
     const {
         countryCodeField = 'country',
+        field = 'region',
         optionValueKey = 'code',
         queries: { getRegionsQuery }
     } = props;
 
+    const hasInitialized = useRef(false);
     const countryFieldState = useFieldState(countryCodeField);
     const { value: country } = countryFieldState;
+    const regionFieldApi = useFieldApi(field);
+
+    // Reset region value when country changes. Because of how Informed sets initialValues,
+    // we want to skip the first state change of the value being initialized.
+    useEffect(() => {
+        if (country) {
+            if (hasInitialized.current) {
+                regionFieldApi.setValue();
+            } else {
+                hasInitialized.current = true;
+            }
+        }
+    }, [country, regionFieldApi]);
 
     const { data, error, loading } = useQuery(getRegionsQuery, {
         variables: { countryCode: country }
     });
 
-    let formattedRegionsData = [];
+    let formattedRegionsData = [{ label: 'Loading Regions...', value: '' }];
```

Comment:
Do we have a test for this? Looks like it is being manipulated in multiple places. A set of tests can be of help.

I see you have added snapshot tests but more verbal ones can be helpful.

Reply:
Added in ab0adc9